### DISCUSSION TITLE
VectorStyleRenderer cannot render feature warning, for Raster feature

### DISCUSSION THREAD

**Discussion by jameslavery** [Mapsui/Mapsui#2233] [Q&A] [answered]
I'm successfully displaying a GeoTiff, using code gleaned from the GeoTiffSample.

The MRaster is created using the following code:

```
using var data = ReadImageAsStream(tiffPath, noDataColors);

        _mRaster = new MRaster(data.ToArray(), _extent);
        _feature = new RasterFeature(_mRaster);
        _feature.Styles.Add(new RasterStyle());
```

However, with the latest version of Mapsui I'm getting this warning (which I can see was added in the recent release) from VectorStyleRenderer:

 >Warning: 0 : VectorStyleRenderer can not render feature of type 'Mapsui.Layers.RasterFeature', Layer: 2036-0

Why am I getting this warning? Why is the VectorStyleRenderer being called for my Raster feature? Have I missed something in creating my Raster layer?

**Comment by charlenni** [comment, marked as answer]
Are there any Styles in the Layer.Styles or in Feature.Styles?

**Reply by jameslavery** [reply]
Thanks - that was the problem. The Layer constructor was creating a VectorStyle which I don't need. Setting the Style to null fixed the problem:

```
return new Layer
        {
            Name = name,
            DataSource = gifSource,
            Enabled = true,
            Style = null
        };
```
